### PR TITLE
[FEATURE] Update documentation on hooks

### PR DESCRIPTION
**Pull request by mk-mxp**
Add missing documentation updates for #338 and #408

### CODE REVIEW

**Comment by mk-mxp** [comment]
Before doubling the work with translations, please confirm this is the right way to go.

**Comment by mk-mxp** [comment]
I'd continue here (add registration form hook) before sending in more hooks. I want to be able to add the docs right with the hook's PR. Is this the way to go?

**Comment by oliverklee** [comment]
Yes, it is.

**Comment by mk-mxp** [comment]
Completed with registration form hook and German.

**Comment by mk-mxp** [comment]
Is there an easy way to test rendering of the docs?

**Comment by oliverklee** [comment]
> Is there an easy way to test rendering of the docs?

From worst to best: On GitHub, in PhpStorm, or using the docker container for building TYPO3 documentation (of which I don't know the details).

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 1]
```diff
@@ -16,115 +16,318 @@
 Hooks
 ^^^^^
 
+.. warning::
+    Um Hooks nutzen zu können, benötigen Sie tiefgehende Kenntnisse über PHP-Klassen,
+    die Implementation von Interfaces und der seminars Objekt Interna.
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: seminars Objekt Interna -> seminars-Objekt-Interna.

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 2]
```diff
@@ -16,115 +16,318 @@
...
 
-New hooks for the single view
-"""""""""""""""""""""""""""""
+Mit Hilfe von Hooks erweitern Sie die Funktionalität von seminars ohne XCLASSen zu nutzen.
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: Please add a comma after "seminars".

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 3]
```diff
@@ -16,115 +16,318 @@
...
+* :ref:`backendemail`
 
-::
+Bitte nehmen Sie Kontakt zu uns auf, wenn Sie weitere Hooks benötigen.
 
-   $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars']['singleView'][]
-         = \Tx_Seminarspaypal_Hooks_EventSingleView::class;
+.. important::
+    seminars wird derzeit grundlegend überarbeitet, um es an die Weiterentwicklung von TYPO3
+    Programmiertechniken anzupassen. Wir bemühen uns, die Änderungen so gering wie möglich zu
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: TYPO3 Programmiertechniken -> TYPO3-Programmiertechniken

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 4]
```diff
@@ -16,115 +16,318 @@
...
+    halten. Informieren Sie sich über Änderungen in CHANGELOG.md, den DocBlocks der Interfaces
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: Please add a comma after "Interfaces".

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 5]
```diff
@@ -16,115 +16,318 @@
...
+    :php:`\OliverKlee\Seminars\Hooks\Interfaces\SeminarSingleView`.
 
-/\*\*
+Es gibt einen Hook in die Einzelansicht. Er wird aufgerufen, bevor das Template zu HTML
+umgewandelt wird. Sie können damit eigene Marker ausfüllen oder bestehende Marker-Werte 
+verändern. Für Details zu Eigenschaften und Methoden siehe :file:`Classes/Frontend/DefaultController.php`.
 
-\* Modifies the event single view.
+Ihre Klasse, die :php:`\OliverKlee\Seminars\Hooks\Interfaces\SeminarSingleView` implementiert,
+machen Sie seminars in :file:`ext_localconf.php` Ihrer Extension bekannt:: php
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: Please use `.. code-block:: php` here (like this: https://docs.typo3.org/m/typo3/docs-how-to-document/master/en-us/WritingReST/CheatSheet.html#code-block-directive).

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 6]
```diff
@@ -16,115 +16,318 @@
...
-\*
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarSingleView;
+    $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][SeminarSingleView:class][]
+        = \Tx_Seminarspaypal_Hooks_EventSingleView::class;
 
-\*  **@param** tx\_seminars\_Model\_Event $event
+Implementieren Sie die benötigten Methoden gemäß dem Interface:: php
 
-\* the event to display in the single view
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarSingleView;
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: Ditto.

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 7]
```diff
@@ -16,115 +16,318 @@
...
+* Bevor der Tabellenfuß in HTML umgewandelt wird
 
-\* Modifies a list row in the time slots list (which is part of the
-event
+In diesen Hooks können Sie eigene Marker ausfüllen oder vorhandene Marker-Werte ändern. Zu
+verfügbaren Eigenschaften und Methoden siehe :file:`Classes/Frontend/DefaultController.php`.
 
-\* single view).
+Der Hook in die Erstellung der Seminar- oder Registrierungen-Bag erlaubt es, die für die Liste
+ausgewählten Seminare / Reqistrierungen zu beeinflussen. Sie können neue Filter hinzufügen oder
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: Please drop the spaces around the slash.

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 8]
```diff
@@ -16,115 +16,318 @@
...
+* Termine am nächsten Tag (`events_next_day`)
+* Andere Termine (`other_dates`)
 
-\* the time slot to display in the current row
+Die letzten beiden Listenarten (Termine am nächsten Tag und Andere Termine) gehören zur Einzelansicht,
+werden aber als komplette Listenansicht behandelt (inklusive Erstellung der Seminar-Bag).
 
-\*  **@param** Tx\_Oelib\_Template $template
+Ihre Klasse, die :php:`\OliverKlee\Seminars\Hooks\Interfaces\SeminarListView` implementiert,
+machen Sie seminars in :file:`ext_localconf.php` Ihrer Extension bekannt:: php
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: Please use code-block here.

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 9]
```diff
@@ -16,115 +16,318 @@
...
+Es bearbeitet sowohl Registrierung alsauch Abmeldung auf einer oder 2 Formularseiten gemäß der Konfiguration.
+Je nachdem, auf welcher Seite man sich befindet bzw. ob es sich um eine Abmeldung handelt, sind bereits Formularwerte
+eingetragen oder nicht. Wenn Sie eigene Formularfelder hinzufügen wollen, müssen Sie die Speicherung in die DB und das
+Ausfüllen passend zu Seite und Status innerhalb des Registrierungs- bzw. Abmeldeprozesses behandeln. Eine Validierung
+erfolgt bei Bedarf durch `mkforms` (nicht über diese Hooks).
+
+Ihre Klasse, die :php:`\OliverKlee\Seminars\Hooks\Interfaces\SeminarRegistrationForm` implementiert,
+machen Sie seminars in :file:`ext_localconf.php` Ihrer Extension bekannt:: php
+
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarRegistrationForm;
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: code-block, please.

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 10]
```diff
@@ -16,115 +16,318 @@
...
+    $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][SeminarRegistrationForm:class][]
+        = \Tx_Seminarspaypal_Hooks_SeminarRegistrationForm::class;
+
+Implementieren Sie die benötigten Methoden gemäß dem Interface:: php
+
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarRegistrationForm;
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: code-block, please.

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 11]
```diff
@@ -204,6 +407,37 @@ example:
...
+array('dear' => String, 'title' => String, 'name' => String)
+
+The second parameter is an user object \Tx_Seminars_Model_FrontEndUser.
+
+Your class then needs to be included and registered like in this
+example:
+
+::
+
+   // register my hook objects
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: code-block

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 12]
```diff
@@ -16,116 +16,319 @@
...
+* :ref:`backendemail`
 
-::
+Please contact us if you need additional hooks.
 
-   $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars']['singleView'][]
-         = \Tx_Seminarspaypal_Hooks_EventSingleView::class;
+.. important::
+    seminars is undergoing a major rewriting to keep up with modern TYPO3 programming
+    techniques. We try to keep changes as small as possible. Inform Yourself about changes
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst]: Inform -> Please inform

Yourself -> yourself

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 13]
```diff
@@ -16,116 +16,319 @@
...
+    by reading CHANGELOG.md, the DocBlocks of interfaces You implement and this
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst]: You -> you

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 14]
```diff
@@ -16,116 +16,319 @@
...
 
-\*
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarSingleView;
+    $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][SeminarSingleView:class][]
+        = \Tx_Seminarspaypal_Hooks_EventSingleView::class;
 
-\*  **@param** tx\_seminars\_Model\_Event $event
+Implement the methods required by the interface:: php
 
-\* the event to display in the single view
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst]: code-block, please.

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 15]
```diff
@@ -16,116 +16,319 @@
...
-\* the time slot to display in the current row
+The last two list types (events next day and other dates) are part of the single
+view, but handled as fully rendered seminar lists (including bag building).
 
-\*  **@param** Tx\_Oelib\_Template $template
+Register Your class that implements :php:`\OliverKlee\Seminars\Hooks\Interfaces\SeminarListView`
+like this in :file:`ext_localconf.php` of Your extension:: php
 
-\* the template that will be used to create the list row output
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarListView;
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst]: code-block, please.

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 16]
```diff
@@ -16,116 +16,319 @@
...
+    $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][SeminarListView:class][]
+        = \Tx_Seminarspaypal_Hooks_ListView::class;
 
-\*
+Implement the methods required by the interface:: php
 
-\*  **@return** void
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarListView;
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst]: code-block, please.

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 17]
```diff
@@ -16,116 +16,319 @@
...
+It handles the registration or unregistration in 1 or 2 pages according to configuraton. Depending on
+the page shown, the previously entered values and if it is an unregistration or not the values in the
+form may be set or not. If You add custom fields to the form You also need to handle storage and
+retrieval in DB for them according to the page / state of the (un)registration process as well as
+validation via `mkforms`.
+
+Register Your class that implements :php:`\OliverKlee\Seminars\Hooks\Interfaces\SeminarRegistrationForm`
+like this in :file:`ext_localconf.php` of Your extension:: php
+
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarRegistrationForm;
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst]: code-block, please.

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 18]
```diff
@@ -16,116 +16,319 @@
...
+
+The registration form is rendered by the builder class in :file:`Classes/Frontend/RegistrationForm.php`.
+It handles the registration or unregistration in 1 or 2 pages according to configuraton. Depending on
+the page shown, the previously entered values and if it is an unregistration or not the values in the
+form may be set or not. If You add custom fields to the form You also need to handle storage and
+retrieval in DB for them according to the page / state of the (un)registration process as well as
+validation via `mkforms`.
+
+Register Your class that implements :php:`\OliverKlee\Seminars\Hooks\Interfaces\SeminarRegistrationForm`
+like this in :file:`ext_localconf.php` of Your extension:: php
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst]: Your -> your (generally)

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 19]
```diff
@@ -16,116 +16,319 @@
...
+
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarRegistrationForm;
+    $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][SeminarRegistrationForm:class][]
+        = \Tx_Seminarspaypal_Hooks_SeminarRegistrationForm::class;
+
+Implement the methods required by the interface:: php
+
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarRegistrationForm;
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst]: code-block, please.

**Comment by mk-mxp** [comment]
Did You find time to proof-read this? I'd like to add 7 more hooks, but include the docs into the hook's pull requests (not here).

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 20]
```diff
@@ -16,115 +16,330 @@
...
 
-\* Modifies the event single view.
+Ihre Klasse, die :php:`\OliverKlee\Seminars\Hooks\Interfaces\SeminarSingleView` implementiert,
+machen Sie seminars in :file:`ext_localconf.php` Ihrer Extension bekannt:
 
-\*
+.. code-block:: php
 
-\*  **@param** tx\_seminars\_Model\_Event $event
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarSingleView;
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: For the hook registration (which happens in ext_localconf.php etc.), we cannot use classes AFAIK, and need to use FQCN.

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 21]
```diff
@@ -16,115 +16,330 @@
...
 
-\*
+Ihre Klasse, die :php:`\OliverKlee\Seminars\Hooks\Interfaces\SeminarListView` implementiert,
+machen Sie seminars in :file:`ext_localconf.php` Ihrer Extension bekannt:
 
-\*  **@return** void
+.. code-block:: php
 
-\*/
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarListView;
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: For the hook registration (which happens in ext_localconf.php etc.), we cannot use classes AFAIK, and need to use FQCN.

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 22]
```diff
@@ -16,115 +16,330 @@
...
+eingetragen oder nicht. Wenn Sie eigene Formularfelder hinzufügen wollen, müssen Sie die Speicherung in die DB und das
+Ausfüllen passend zu Seite und Status innerhalb des Registrierungs- bzw. Abmeldeprozesses behandeln. Eine Validierung
+erfolgt bei Bedarf durch `mkforms` (nicht über diese Hooks).
+
+Ihre Klasse, die :php:`\OliverKlee\Seminars\Hooks\Interfaces\SeminarRegistrationForm` implementiert,
+machen Sie seminars in :file:`ext_localconf.php` Ihrer Extension bekannt:
+
+.. code-block:: php
+
+    use \OliverKlee\Seminars\Hooks\Interfaces\SeminarRegistrationForm;
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst]: For the hook registration (which happens in ext_localconf.php etc.), we cannot use classes AFAIK, and need to use FQCN.

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 23]
```diff
@@ -16,115 +16,330 @@
...
+         *
+         * This function will be called for all types of seminar lists ("topics",
+         * "seminars", "my seminars", "my vip seminars", "my entered events",
+         * "events next day", "other dates").
+         *
+         * @param \Tx_Seminars_FrontEnd_DefaultController $controller the calling controller
+         *
+         * @return void
+         */
+        public function modifyListFooter(\Tx_Seminars_FrontEnd_DefaultController $controller)
```
[the remark this answers is not in the document]
Comment (mk-mxp) [reply Documentation/DE/Referenz/Hooks/Index.rst]: I don't see the additional space. It is exactly the same style as used all around the whole project (and in your own latest file additions): 4 spaces for the code block, 4 spaces for the class block, start of DocBlock `/**` no additional space, rest of DocBlock including `*/` 1 addiitional space. Also it renders fine. Did I miss something?

[the remark this answers is not in the document]
Comment (oliverklee) [reply Documentation/DE/Referenz/Hooks/Index.rst]: I was not looking straight, and the code actually is okay. Sorry for the confusion.

**Pull request merged**